### ISSUE TITLE
Zeroing out the gradient before calling optimizer.step() in train_eval_seg.py

### ISSUE THREAD

**Issue by mitalbert** [sacmehta/EdgeNets#14]
Hi, thanks for publishing the code.

I noticed the optimizer.zero_grad() is called after calling optimizer.step() in train_eval_seg.py. Is this intentional? I was having issues until I moved zero_grad() above loss.backward().

Also when resuming the training there was a runtime error with type compatibility (cpu vs cuda) when calculating loss.backward() and adding model = model.cuda() right after loading the state  dictionary in train_segmentation.py solved it.

**Comment by mitalbert** [comment]
Sure, I'm using 1.1.0

**Comment by sacmehta** [comment]
@mitalbert I have incorporated your suggestions. Please check if it resolves your issues.

**Comment by mitalbert** [comment]
If you mean the first issue in train_eval_seg.py, then it looks exactly like my fix with worked for me, so, yes.

If you mean the issue with resuming the training, unfortunately I can't test it now.
